Added 4 background-size (with no intrinsic and no ratio) tests

TalbotG opened the pull request:
This PR spun from 
https://github.com/web-platform-tests/wpt/pull/22201
and is a followup to
[[css-backgrounds-3] background-size with " auto" and gradient image is not interoperably implemented](https://github.com/w3c/csswg-drafts/issues/2675)

Here are the last review comments from #22201 :
{
Ways this test can pass on an incorrect implementation:

- UA doesn't support background-size at all, or ignores the declaration.
- UA handles sizing in the y axis correctly, but not the x axis.
- UA resolves to a size taller than the background positioning area. (We can't tell if the image was sized correctly, or oversized and clipped to the background painting area.)
- UA assumes a 1:1 ratio -- such an incorrect assumption will still pass 4/6 subtests.
} 

Now the 4 (041 to 044) tests in here 
- declare 'background-repeat: no-repeat'
- use a 2-color image (SVG image and linear-gradient)
- use 'background-position: center'
- have 'background-origin' declaration to be different from 'background-clip'
- have 'background-origin' and 'background-clip' do not use their initial respective value
- will fail if the UA doesn't support background-size at all, or ignores the declaration

On my website, these 4 (041 to 044) tests are:

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/background-size-061.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/reference/background-size-061-ref.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/background-size-062.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/reference/background-size-062-ref.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/background-size-063.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/reference/background-size-063-ref.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/background-size-064.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Backgrounds/reference/background-size-064-ref.html

TalbotG committed:
Added 4 background-size (with no intrinsic and no ratio) tests

fantasai commented:
A few things to fix:
* Use `background-clip: border-box; background-origin: content-box` instead of the other way around so that we see the whole image. Particularly for the SVG, it's not possible to see if it is stretching correctly otherwise. Use a `transparent` border so that the background is visible in the border area.
* The calculations on the border for the SVG are not quite right, so the image and the reference don't match.

Otherwise looks pretty good.

Note: “have 'background-origin' and 'background-clip' do not use their initial respective value” -- this isn't necessary, the only thing that's necessary is that the two boxes are not the same so that we make sure the UA is using the correct one.

fantasai reviewed (changes requested):
See above.

TalbotG committed:
Corrected background-size-04[1-4] according to review

TalbotG commented:
In the 10385a2 patch, I have
- used `background-clip: border-box; background-origin: content-box`
- used transparent borders for top and bottom borders
- renamed div#black-border as div#black-walls
- added an empty line in support/orange-intrinsic-none.svg at EOF
- removed reference/background-size-042-ref.html and reference/background-size-044-ref.html; the 4 tests now only require 2 reference files
